samples/Client/DaprClient/StateStoreETagsExample.cs

```diff
@@ -21,30 +21,63 @@ public override async Task RunAsync(CancellationToken cancellationToken)
+            Console.WriteLine($"Saved state with new etag");
+
+            // Modify the state and try saving it with the old etag. This will fail
+            original.Color = "purple";
+            var isSaveStateSuccess = await client.TrySaveStateAsync<Widget>(storeName, stateKeyName, original, originalETag, cancellationToken: cancellationToken);
+            Console.WriteLine($"Saved state with old etag: {originalETag} successfully? {isSaveStateSuccess}");
 
             // Read the updated state and etag
             var (updated, updatedETag) = await client.GetStateAndETagAsync<Widget>(storeName, stateKeyName, cancellationToken: cancellationToken);
             Console.WriteLine($"Retrieved state: {updated.Size}  {updated.Color} with etag: {updatedETag}");
 
-            // Modify the state and try saving it with the old etag. This will fail
-            updated.Color = "purple";
-            var isSaveStateSuccess = await client.TrySaveStateAsync<Widget>(storeName, stateKeyName, updated, originalETag, cancellationToken: cancellationToken);
-            Console.WriteLine($"Saved state with old etag: {originalETag} successfully? {isSaveStateSuccess}");
-
             // Modify the state and try saving it with the updated etag. This will succeed
+            updated.Color = "green";
             isSaveStateSuccess = await client.TrySaveStateAsync<Widget>(storeName, stateKeyName, updated, updatedETag, cancellationToken: cancellationToken);
             Console.WriteLine($"Saved state with latest etag: {updatedETag} successfully? {isSaveStateSuccess}");
+
+            // Delete with a null etag. This will throw an ArgumentException
+            try
+            {
+                await client.TryDeleteStateAsync(storeName, stateKeyName, null, cancellationToken: cancellationToken);
+            }
+            catch(ArgumentException)
+            {
+                Console.WriteLine($"TryDeleteStateAsync with null etag threw ArgumentException");
+            }
+
+            // Delete with an old etag. This will fail
+            var isDeleteStateSuccess = await client.TryDeleteStateAsync(storeName, stateKeyName, originalETag, cancellationToken: cancellationToken);
+            Console.WriteLine($"Deleted state with old etag: {originalETag} successfully? {isDeleteStateSuccess}");
+
+            // Read the updated state and etag
+            (updated, updatedETag) = await client.GetStateAndETagAsync<Widget>(storeName, stateKeyName, cancellationToken: cancellationToken);
+            Console.WriteLine($"Retrieved state: {updated.Size}  {updated.Color} with etag: {updatedETag}");
+
+            // Delete with updated etag. This will succeed
+            isDeleteStateSuccess = await client.TryDeleteStateAsync(storeName, stateKeyName, updatedETag, cancellationToken: cancellationToken);
+            Console.WriteLine($"Deleted state with updated etag: {updatedETag} successfully? {isDeleteStateSuccess}");
```

Comment:
Can you explain what the changes are to this sample? There are changes here, but I'm not really sure how it makes this a better sample.

Reply:
I added the TryDelete API because realized that the earlier sample only used Save.